src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -464,8 +464,9 @@ void VideoDecoder::addVideoStreamDecoder(
   }
 
   if (options.device.type() == torch::kCUDA) {
-    codec = findCudaCodec(options.device, streamInfo.stream->codecpar->codec_id)
-                .value_or(codec);
+    codec = makeAVCodecPtrBestStream(
+        findCudaCodec(options.device, streamInfo.stream->codecpar->codec_id)
+            .value_or(codec));
```

Comment:
Just to check my understanding: the main point of this PR is that previously, `findCudaCodec` would always return a `const AVCodec*`, for all FFmpeg versions. This  `const AVCodec*` would be cast to a `AVCodecPtr`, and on some FFmpeg versions that would implicitly become a non-const `AVCodec*`? By calling `makeAVCodecPtrBestStream`, now we're making this cast explicit?

Reply:
Actually, no, previously `findCudaCodec()` returned an `AVCodecPtr`. But inside of `findCudaCodec` we were actually looking for and finding what was always a `const AVCodec*`. The FFmpeg functions we're calling in that function always take and return a `const AVCodec*`. That caused a problem with FFmpeg version 4, because in that version, `AVCodecPtr` is just a `AVCodec*`. The problem is we would try to assign a const pointer to a non-const pointer, which is not an allowed implicit conversion. The `makeAVCodecPtrBestStream()` utility does the appropriate const-cast when needed.